### PR TITLE
 dev into global 10/29

### PR DESCRIPTION
**Pull request by angaither**
Everybody's favorite rebase! HOPEFULLY THIS IS THE LAST TIME WE DO THIS. 

note, these were the conflicts I fixed
@sergii-tkachenko @blisteringherb if you want to double check those files
### Conflicts:
- [x] lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module
- [x] lib/modules/dosomething/dosomething_global/dosomething_global.info
- [x] lib/modules/dosomething/dosomething_global/dosomething_global.strongarm.inc
- [x]  lib/modules/dosomething/dosomething_signup/dosomething_signup.module

### CODE REVIEW

**Comment by sergiitk** [comment]
Just to document the reason of `dosomething_global.strongarm.inc` merge conflict, see #5661:

> This PR will cause merge conflict of `dosomething_global.strongarm.inc` with global branch.
> Please solve it in favor of global's code.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module [hunk 1]
```diff
@@ -568,14 +568,22 @@ function dosomething_signup_get_mbp_params($account, $node, $opt_in) {
...
   $params = array(
     'email' => $account->mail,
     'uid' => $account->uid,
     'first_name' => dosomething_user_get_field('field_first_name', $account),
     'mobile' => dosomething_user_get_field('field_mobile', $account),
     'event_id' => $node->nid,
-    'campaign_title' => $wrapper->language($language)->title_field->value(),
-    'campaign_link' => dosomething_global_url('node/' . $node->nid, $url_options),
+    'campaign_title' => $title,
+    'campaign_link' => url('node/' . $node->nid, $url_options),
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: Not sure about this line.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module [hunk 2]
```diff
@@ -568,14 +568,22 @@ function dosomething_signup_get_mbp_params($account, $node, $opt_in) {
...
+
   $params = array(
     'email' => $account->mail,
     'uid' => $account->uid,
     'first_name' => dosomething_user_get_field('field_first_name', $account),
     'mobile' => dosomething_user_get_field('field_mobile', $account),
     'event_id' => $node->nid,
-    'campaign_title' => $wrapper->language($language)->title_field->value(),
-    'campaign_link' => dosomething_global_url('node/' . $node->nid, $url_options),
+    'campaign_title' => $title,
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module:585]: Good.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module [hunk 3]
```diff
@@ -568,14 +568,22 @@ function dosomething_signup_get_mbp_params($account, $node, $opt_in) {
   );
   $campaign_link = dosomething_global_url('node/' . $nid, $url_options);
 
+  // Get node title, normal for collections, translatable for campaigns.
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module:571]: Good.

File: lib/modules/dosomething/dosomething_global/dosomething_global.strongarm.inc [hunk 4]
```diff
@@ -112,6 +112,7 @@ function dosomething_global_strongarm() {
       ),
     ),
   );
+
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_global/dosomething_global.strongarm.inc:115]: Good.

File: lib/modules/dosomething/dosomething_global/dosomething_global.info [hunk 5]
```diff
@@ -27,4 +27,4 @@ features[variable][] = pathauto_node_campaign_en-global_pattern
 features[variable][] = pathauto_node_campaign_en_pattern
 features[variable][] = pathauto_node_campaign_es-mx_pattern
 features[variable][] = pathauto_node_campaign_pt-br_pattern
-mtime = 1443727804
+mtime = 1445972087
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_global/dosomething_global.info:30]: Good.

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 6]
```diff
@@ -541,8 +541,7 @@ function dosomething_campaign_get_campaigns_doing($uid = NULL) {
  */
 function dosomething_campaign_get_campaign_block_vars($nid, $image_size = '740x480', $source = NULL) {
   $node = node_load($nid);
-  $clc = dosomething_helpers_get_current_language_code();
-
+  $clc = dosomething_helpers_get_current_language_content_code();
```
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module:544]: Good.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module (hunk 1 above)
Comment (sergiitk) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: I think this one should be `dosomething_global_url()`
See https://github.com/angaither/dosomething/commit/15f7281096e5ee0ef690a1c5de5511f87688dbf9
cc @blisteringherb

**Comment by angaither** [comment]
@sergii-tkachenko which resolution is [incorrect](https://github.com/DoSomething/phoenix/pull/5663#issuecomment-152217631) (besides the signup code)

**Comment by sergiitk** [comment]
@angaither I just copied the note from PR with the comment how to resolve `dosomething_global.strongarm.inc` for sake of documenting it.

**Comment by sergiitk** [comment]
Everything looks correct, besides this: https://github.com/DoSomething/phoenix/pull/5663#discussion_r43401663

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.module (hunk 1 above)
Comment (blisteringherb) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.module]: It should be `dosomething_global_url()`

**Comment by sergiitk** [comment]
:+1:

**Pull request merged**